plugins/c9.ide.download/download.js

```diff
@@ -55,25 +82,40 @@ define(function(require, exports, module) {
             if (node.isFolder && node.path == "/")
                 downloadProject();
             else if (paths.length > 1)
-                vfs.download(paths);
+                downloadPaths(paths);
             else if (node.isFolder)
                 downloadFolder(node.path);
             else
                 downloadFile(node.path);
-            
+
         }
-        
+
         function downloadProject() {
-            vfs.download("/", info.getWorkspace().name + ".tar.gz");
+            vfs.download("/", info.getWorkspace().name + getArchiveFileExtension());
+        }
+
+        function downloadPaths(paths) {
+            vfs.download(paths, info.getWorkspace().name + getArchiveFileExtension());
```

Comment:
I am worried that returning workspace name can be misleading too, e.g. user might think whole workspace is backed up, delete it, and find out that only a couple of files were backed up.